### DISCUSSION TITLE
how to read a image to make a network request?

### DISCUSSION THREAD

**Discussion by WingDust** [nushell/nushell#9772] [Q&A]
`open aa.png | encode base64 | curl -x POST http://localhost:8000  -H "Content-Type: application/json"  -d '{data:"how to write"}' `

**Comment by ghost** [comment]
as far as i know you just use curl for it without anything else.  
you can pass a file by using `@aa.png` as content (example: `curl -X POST http://localhost:8000 -H "Content-Type: application/json" -d @aa.png` or `curl -X POST http://localhost:8000 -H "Content-Type: application/json" -F file=@aa.png`)
([here are some real examples](https://0x0.st/))

if you want to use nushell without any external programs you could try `http post -H ["Content-Type" "application/json"] http://localhost:8000 (open -r aa.png)`

although im not sure why you would want to tag a png as json
